src/bezpath.rs

```diff
@@ -1648,4 +1660,360 @@ mod tests {
+                PathEl::CurveTo((5.0, 5.0).into(), (6.0, 6.0).into(), (7.0, 7.0).into()),
+                PathEl::ClosePath,
+            ],
+            vec![
+                PathEl::MoveTo((7.0, 7.0).into()),
+                PathEl::CurveTo((6.0, 6.0).into(), (5.0, 5.0).into(), (4.0, 4.0).into()),
+                PathEl::CurveTo((3.0, 3.0).into(), (2.0, 2.0).into(), (1.0, 1.0).into()),
+                PathEl::LineTo((0.0, 0.0).into()), // ... does NOT become implied
+                PathEl::ClosePath,
+            ],
+        )
+    }
+
+    #[test]
+    fn test_reverse_closed_last_quad_overlaps_move() {
+        reverse_test_helper(
+            vec![
+                PathEl::MoveTo((0.0, 0.0).into()),
+                PathEl::QuadTo((1.0, 1.0).into(), (2.0, 2.0).into()),
+                PathEl::QuadTo((3.0, 3.0).into(), (0.0, 0.0).into()),
+                PathEl::ClosePath,
+            ],
+            vec![
+                PathEl::MoveTo((0.0, 0.0).into()), // no extra lineTo added here
+                PathEl::QuadTo((3.0, 3.0).into(), (2.0, 2.0).into()),
+                PathEl::QuadTo((1.0, 1.0).into(), (0.0, 0.0).into()),
+                PathEl::ClosePath,
+            ],
+        )
+    }
+
+    #[test]
+    fn test_reverse_closed_last_quad_not_on_move() {
+        reverse_test_helper(
+            vec![
+                PathEl::MoveTo((0.0, 0.0).into()),
+                PathEl::QuadTo((1.0, 1.0).into(), (2.0, 2.0).into()),
+                PathEl::QuadTo((3.0, 3.0).into(), (4.0, 4.0).into()),
+                PathEl::ClosePath,
+            ],
+            vec![
+                PathEl::MoveTo((4.0, 4.0).into()), // the previously implied line
+                PathEl::QuadTo((3.0, 3.0).into(), (2.0, 2.0).into()),
+                PathEl::QuadTo((1.0, 1.0).into(), (0.0, 0.0).into()),
+                PathEl::ClosePath,
+            ],
+        )
+    }
+
+    #[test]
+    fn test_reverse_closed_line_quad_line() {
+        reverse_test_helper(
+            vec![
+                PathEl::MoveTo((0.0, 0.0).into()),
+                PathEl::LineTo((1.0, 1.0).into()), // this line...
+                PathEl::QuadTo((2.0, 2.0).into(), (3.0, 3.0).into()),
+                PathEl::ClosePath,
+            ],
+            vec![
+                PathEl::MoveTo((3.0, 3.0).into()),
+                PathEl::QuadTo((2.0, 2.0).into(), (1.0, 1.0).into()),
+                PathEl::LineTo((0.0, 0.0).into()), // ... does NOT become implied
+                PathEl::ClosePath,
+            ],
+        )
+    }
+
+    #[test]
+    fn test_reverse_empty() {
+        reverse_test_helper(vec![], vec![])
+    }
+
+    #[test]
+    fn test_reverse_single_point() {
+        reverse_test_helper(
+            vec![PathEl::MoveTo((0.0, 0.0).into())],
+            vec![PathEl::MoveTo((0.0, 0.0).into())],
+        )
+    }
+
+    #[test]
+    fn test_reverse_single_point_closed() {
```

Comment:
this test and the one immediately preceding are exactly the same as test_reverse_move_only_path_offset_validity and test_reverse_move_close_path_offset_validity, are they not?

Reply:
They appear to be, but I just copied them all directly from our tests in write-fonts :) I'll remove the redundant ones.